airsonic-main/src/main/java/org/airsonic/player/dao/CoverArtDao.java

```diff
@@ -51,9 +51,9 @@ public void delete(EntityType type, int id) {
 
     @Transactional(isolation = Isolation.READ_COMMITTED)
     public void expunge() {
-        update("delete from cover_art where entity_type='MEDIA_FILE' and entity_id in (select ca.entity_id from cover_art ca left join media_file m on ca.entity_id=m.id where m.id is null and ca.entity_type='MEDIA_FILE')");
-        update("delete from cover_art where entity_type='ALBUM' and entity_id in (select ca.entity_id from cover_art ca left join album a on ca.entity_id=a.id where a.id is null and ca.entity_type='ALBUM')");
-        update("delete from cover_art where entity_type='ARTTIST' and entity_id in (select ca.entity_id from cover_art ca left join artist a on ca.entity_id=a.id where a.id is null and ca.entity_type='ARTIST')");
+        update("DELETE FROM cover_art WHERE entity_type='MEDIA_FILE' AND entity_id IN (SELECT media_file.id FROM media_file WHERE media_file.id IS NULL)");
```

Comment:
this is incorrect and not equivalent

cover_art table has a column called entity_id which is, for entity_type='MEDIA_FILE', populated by ids that are rows in media_file table (media_file.id)

so imagine a media_file:
| id | path | ... |
|---|-------|---|
|1  | filea.mp3 | - |
|2 | fileb.mp3 | - |

with its cover_art
| entity_id |entity_type| ... |
|---|-------|---|
|1  | MEDIA_FILE | - |
|2  | MEDIA_FILE | - |

what ends up happening before this expunge is that we delete some of the media_files. now in this method we're looking to delete the cover_art for which no media_file exists

so imagine from our example earlier, if the row with id 1 gets deleted from media_files
| id | path | ... |
|---|-------|---|
|2 | fileb.mp3 | - |

and the cover_art is like 
| entity_id | type| ... |
|---|-------|---|
|1  | MEDIA_FILE | - |
|2 | MEDIA_FILE | - |

how do we delete this cover art?

- we left-join cover_art with media_file on ids, that'll yield:

| entity_id | type| id | path |
|----------|-------|---|------|
|1  | MEDIA_FILE | null | null |
|2 | MEDIA_FILE | 2 | fileb.mp3 |

and we can delete everything that has a null id.

- `SELECT media_file.id FROM media_file WHERE media_file.id IS NULL` will yield nothing because id is a primary key for media_file, it cannot be null